http-client-core/src/main/java/io/micronaut/http/client/bind/DefaultHttpClientBinderRegistry.java

```diff
@@ -0,0 +1,164 @@
+                    .stringValue(RequestAttribute.class)
+                    .filter(StringUtils::isNotEmpty)
+                    .orElse(NameUtils.hyphenate(context.getArgument().getName()));
+            request.getAttributes().put(attributeName, value);
+        });
+        byAnnotation.put(Body.class, (ClientArgumentRequestBinder<Object>) (context, value, request) -> {
+            request.body(value);
+        });
+
+        if (CollectionUtils.isNotEmpty(binders)) {
+            for (ClientArgumentRequestBinder binder : binders) {
+                addBinder(binder);
+            }
+        }
+    }
+
+    @Override
+    public <T> Optional<ClientArgumentRequestBinder<T>> findArgumentBinder(Argument<T> argument) {
+        Optional<Class<? extends Annotation>> opt = argument.getAnnotationMetadata().getAnnotationTypeByStereotype(Bindable.class);
+        if (opt.isPresent()) {
+            Class<? extends Annotation> annotationType = opt.get();
+            ClientArgumentRequestBinder<T> binder = byAnnotation.get(annotationType);
+            return Optional.ofNullable(binder);
+        } else {
+            ClientArgumentRequestBinder<T> binder = byType.get(argument.typeHashCode());
+            if (binder != null) {
+                return Optional.of(binder);
+            } else {
+                binder = byType.get(Argument.of(argument.getType()).typeHashCode());
+                return Optional.ofNullable(binder);
+            }
+        }
+    }
+
+    /**
+     * Adds a binder to the registry.
+     *
+     * @param binder The binder
+     * @param <T> The type
+     */
+    public <T> void addBinder(ClientArgumentRequestBinder<T> binder) {
+        if (binder instanceof AnnotatedClientArgumentRequestBinder) {
+            AnnotatedClientArgumentRequestBinder<?, ?> annotatedRequestArgumentBinder = (AnnotatedClientArgumentRequestBinder) binder;
+            Class<? extends Annotation> annotationType = annotatedRequestArgumentBinder.getAnnotationType();
+            byAnnotation.put(annotationType, annotatedRequestArgumentBinder);
+        } else if (binder instanceof TypedClientArgumentRequestBinder) {
+            TypedClientArgumentRequestBinder<?> typedRequestArgumentBinder = (TypedClientArgumentRequestBinder) binder;
+            byType.put(typedRequestArgumentBinder.argumentType().typeHashCode(), typedRequestArgumentBinder);
+            List<Class<?>> superTypes = typedRequestArgumentBinder.superTypes();
+            if (CollectionUtils.isNotEmpty(superTypes)) {
+                for (Class<?> superType : superTypes) {
+                    byType.put(Argument.of(superType).typeHashCode(), typedRequestArgumentBinder);
+                }
+            } else if (typedRequestArgumentBinder.supportsSuperTypes()) {
+                Set<Class> allInterfaces = ReflectionUtils.getAllInterfaces(typedRequestArgumentBinder.argumentType().getType());
+                if (ClassUtils.REFLECTION_LOGGER.isWarnEnabled()) {
```

Comment:
We shouldn't be introducing new behaviour that adds warnings

Reply:
Should this be deprecated then in 2.1? https://github.com/micronaut-projects/micronaut-core/blob/2.0.x/http/src/main/java/io/micronaut/http/bind/binders/TypedRequestArgumentBinder.java#L40